core/src/main/java/org/mule/runtime/core/internal/context/ArtifactStoppedListener.java

```diff
@@ -0,0 +1,27 @@
+/*
+ * Copyright (c) MuleSoft, Inc.  All rights reserved.  http://www.mulesoft.com
+ * The software in this package is published under the terms of the CPAL v1.0
+ * license, a copy of which has been included with this distribution in the
+ * LICENSE.txt file.
+ */
+package org.mule.runtime.core.internal.context;
+
+public interface ArtifactStoppedListener {
```

Comment:
the presence of the `mustPersist` method in this interface makes me think that its name is more generic than the intended usage.